In [1]:
import numpy as np

In [2]:
m = np.array([3000, 5000])
frequencies    = np.array([0.1, 0.2, 0.3, 0.4, 0.6])
c_Observation  = np.array([3721, 3191, 3087, 3050, 3023])


c_Theoretical           = np.array([3719.56, 3189.99, 3086.18, 3049.19, 3022.27])
delta_c                 = c_Observation - c_Theoretical
c_Theoretical_pertub_b1 = (np.array([ 3732.75, 3201.52, 3096.93, 3059.63, 3032.47 ]) - c_Theoretical ) / 10.0
c_Theoretical_pertub_b2 = (np.array([ 3721.64, 3190.19, 3086.23, 3049.21, 3022.27 ]) - c_Theoretical ) / 10.0

In [3]:
G = np.column_stack((c_Theoretical_pertub_b1, c_Theoretical_pertub_b2))
GT = np.transpose(G)
GTG = np.matmul ( GT, G)
GTG_inv = np.linalg.inv(GTG)

In [4]:
delta_m_hat = np.matmul( np.matmul(GTG_inv, GT), delta_c)
m_hat = m + delta_m_hat

In [5]:
print(m)
print(m_hat)
print(delta_m_hat)

[3000 5000]
[3000.77222363 5002.06001695]
[0.77222363 2.06001695]


gausschang@Gausss-MacBook-Pro-14 build % ./main

== Theoretical dispersion == by C++

f = 0.1 Hz => c_Theoretical = 3719.56 m/s

f = 0.2 Hz => c_Theoretical = 3189.99 m/s

f = 0.3 Hz => c_Theoretical = 3086.18 m/s

f = 0.4 Hz => c_Theoretical = 3049.19 m/s

f = 0.6 Hz => c_Theoretical = 3022.27 m/s


== Inversion result ==

Updated b1 = 3000.77 m/s

Updated b2 = 5002.04 m/s


== After Pertubation ==

f = 0.1 Hz => c_Theoretical_P = 3721.01 m/s

f = 0.2 Hz => c_Theoretical_P = 3190.92 m/s

f = 0.3 Hz => c_Theoretical_P = 3087.02 m/s

f = 0.4 Hz => c_Theoretical_P = 3050 m/s

f = 0.6 Hz => c_Theoretical_P = 3023.06 m/s

gausschang@Gausss-MacBook-Pro-14 build %
